safe_transaction_service/history/helpers.py

```diff
@@ -56,12 +56,20 @@ def calculate_all_possible_hashes(cls, delegate: ChecksumAddress) -> List[bytes]
 
 def is_valid_unique_transfer_id(unique_transfer_id: str) -> bool:
     """
-    Check if transfer_id starts with 'e' or 'i' followed by keccak256 and appended by optional digits separated by commas
+    Check if transfer_id starts with 'e' or 'i' followed by keccak256 and appended by digits or digits separated by commas
 
     :param unique_transfer_id:
-    :return: True in case valide unique_transfer_id or False in other case
+    :return: ``True`` for a valid ``unique_transfer_id``, ``False`` otherwise
     """
-    return re.match(r"^(e|i)[a-fA-F0-9]{64}(\d+(,\d+)*)?$", unique_transfer_id)
+    token_transfer_id_pattern = r"^(e)([a-fA-F0-9]{64})(\d+)$"
+    internal_transfer_id_pattern = r"^(i)([a-fA-F0-9]{64})(\d+)(,\d+)*$"
+
+    if bool(re.match(token_transfer_id_pattern, unique_transfer_id)) or bool(
+        re.match(internal_transfer_id_pattern, unique_transfer_id)
+    ):
+        return True
+
+    return False
```

Comment:
```suggestion
    return bool(re.match(token_transfer_id_pattern, unique_transfer_id) or re.match(internal_transfer_id_pattern, unique_transfer_id))):
```

No need to cast to `bool` when using `if`, `if` does it automatically, but in this case using `return` is better

Btw, shouldn't we be using `fullmatch`? https://docs.python.org/3/library/re.html#re.Pattern.fullmatch